editor/src/clj/internal/system.clj

```diff
@@ -242,7 +242,11 @@
 
 (defn merge-graphs
   [system post-tx-graphs significantly-modified-graphs outputs-modified nodes-deleted]
-  (let [graph-id->outputs-modified (group-by #(gt/node-id->graph-id (first %)) outputs-modified)
+  (let [graph-id->outputs-modified (util/group-into
+                                     {}
+                                     #{}
+                                     #(gt/node-id->graph-id (gt/endpoint-node-id %))
+                                     outputs-modified)
```

Comment:
We group into sets so there is no need to convert vector to set in `merge-or-push-history`